lib/auditd/auditd_linux.go

```diff
@@ -0,0 +1,288 @@
+/*
+ *
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auditd
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"syscall"
+
+	"github.com/gravitational/trace"
+	"github.com/mdlayher/netlink"
+	"github.com/mdlayher/netlink/nlenc"
+	log "github.com/sirupsen/logrus"
+)
+
+// Client is auditd client.
+type Client struct {
+	conn NetlinkConnecter
+
+	execName     string
+	hostname     string
+	systemUser   string
+	teleportUser string
+	address      string
+	ttyName      string
+
+	mtx     *sync.Mutex
```

Comment:
I wouldn't use a pointer to mutex without reason, you can just use `sync.Mutex` and avoid initialization and having to follow the pointer every time it's accessed. In fact, it's somewhat common to just directly embed `sync.Mutex` into the type.